src/Database/PostgreSQL/PQTypes/SQL/Builder.hs

```diff
@@ -408,9 +415,28 @@ instance Sqlable SqlInsertSelect where
     , emitClausesSepComma "RETURNING" $ sqlInsertSelectResult cmd
     ]
 
+checkAndRememberMaterializationSupport :: (MonadDB m, MonadIO m, MonadMask m) => m ()
```

Comment:
This needs a comment explaining what exactly it does and that it should be called asap after application start.